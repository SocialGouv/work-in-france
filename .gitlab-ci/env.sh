#!/usr/bin/env bash

#

export BRANCH_NAME=${BRANCH_NAME:=$CI_COMMIT_REF_SLUG}
export COMMIT_TAG=${COMMIT_TAG:=$CI_COMMIT_TAG}
export COMMIT=${COMMIT:=$CI_COMMIT_SHA}
export ENVIRONMENT=${ENVIRONMENT:="dev.factory"};
export HASH_SIZE=${HASH_SIZE:=7}
export PROJECT_PATH=${PROJECT_PATH:=$CI_PROJECT_PATH}
export JOB_ID=${JOB_ID:=$CI_JOB_ID}

BRANCH_NAME_HASHED=$( printf "${BRANCH_NAME}" | sha1sum | cut -c1-${HASH_SIZE} )
export BRANCH_HASH=${BRANCH_HASH:=$BRANCH_NAME_HASHED}

#

if [[ "${BRANCH_NAME}" = "master" ]]; then
  export BRANCH_HASH=master;
  export ENVIRONMENT=incubateur;
fi

if [[ -n "${COMMIT_TAG}" ]]; then
  export ENVIRONMENT=incubateur;
  export BRANCH_HASH=$( printf "${COMMIT_TAG}" | sed "s/\./-/g" );
fi

if [[ -n "${PRODUCTION+x}" ]]; then
  export BRANCH_HASH=prod;
  export FRONTEND_HOST="work-in-france.${ENVIRONMENT}.social.gouv.fr";
else
  export FRONTEND_HOST="${BRANCH_HASH}.work-in-france.${ENVIRONMENT}.social.gouv.fr";
fi

#

printenv | grep -E \
  "BRANCH_HASH|BRANCH_NAME|COMMIT|COMMIT_TAG|ENVIRONMENT|HASH_SIZE|JOB_ID" \
  | sort
printenv | grep -E \
  "FRONTEND_HOST" \
  | sort
