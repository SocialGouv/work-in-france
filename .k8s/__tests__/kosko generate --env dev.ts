//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --dev", async () => {
  expect(
    await getEnvManifests("dev", "", {
      ...project("serving-ml").dev,
      RANCHER_PROJECT_ID: "c-bd7z2:p-2qlgk",
      KUBE_NAMESPACE: "work-in-france-50-master-dev2",
    })
  ).toMatchSnapshot();
});
