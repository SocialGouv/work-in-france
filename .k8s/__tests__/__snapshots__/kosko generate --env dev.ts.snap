// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --dev 1`] = `
"---
metadata:
  annotations:
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: 'c-bd7z2:p-2qlgk'
    git/branch: master
    git/remote: >-
      https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/work-in-france.git
    app.gitlab.com/app: socialgouv-work-in-france
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    application: master-dev2-work-in-france
    owner: work-in-france
    team: work-in-france
    cert: wildcard
  name: work-in-france-50-master-dev2
apiVersion: v1
kind: Namespace
---
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-work-in-france
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: www
    application: master-dev2-work-in-france
    owner: work-in-france
    team: work-in-france
    cert: wildcard
    component: nginx
  name: www
  namespace: work-in-france-50-master-dev2
spec:
  replicas: 1
  selector:
    matchLabels:
      app: www
  template:
    metadata:
      annotations:
        app.gitlab.com/app: socialgouv-work-in-france
        app.gitlab.com/env: master-dev2
        app.gitlab.com/env.name: master-dev2
      labels:
        app: www
        application: master-dev2-work-in-france
        owner: work-in-france
        team: work-in-france
        cert: wildcard
        component: nginx
    spec:
      containers:
        - image: >-
            registry.gitlab.factory.social.gouv.fr/socialgouv/work-in-france/www:0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: www
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 32Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /index.html
              port: http
            periodSeconds: 5
apiVersion: apps/v1
kind: Deployment
---
metadata:
  labels:
    app: www
    application: master-dev2-work-in-france
    owner: work-in-france
    team: work-in-france
    cert: wildcard
  name: www
  annotations:
    app.gitlab.com/app: socialgouv-work-in-france
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  namespace: work-in-france-50-master-dev2
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: www
  type: ClusterIP
apiVersion: v1
kind: Service
---
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    app.gitlab.com/app: socialgouv-work-in-france
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: www
    application: master-dev2-work-in-france
    owner: work-in-france
    team: work-in-france
    cert: wildcard
  name: www
  namespace: work-in-france-50-master-dev2
spec:
  rules:
    - host: master-dev2-work-in-france.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              serviceName: www
              servicePort: 80
            path: /
  tls:
    - hosts:
        - master-dev2-work-in-france.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
apiVersion: extensions/v1beta1
kind: Ingress
"
`;
